ops/model.py

```diff
@@ -985,6 +1001,55 @@ def location(self):
         return self._location
 
 
+class Container:
+    """Represents a named container in a unit.
+
+    Attributes:
+        name: The name of the container from metadata.yaml (eg, 'postgres').
+    """
+
+    def __init__(self, name, pebble_client=None):
+        self.name = name
+
+        if pebble_client is None:
+            socket_path = '/charm/containers/{}/pebble/.pebble.socket'.format(name)
+            pebble_client = pebble.PebbleClient(socket_path=socket_path)
+        self._pebble = pebble_client
+
+    @property
+    def pebble(self) -> 'pebble.PebbleClient':
+        """Return the low-level Pebble client instance for this container."""
+        return self._pebble
+
+    def autostart(self):
+        """Autostart all default services."""
+        self._pebble.autostart_services()
+
+    def start(self, *service_names: typing.List[str]):
+        """Start given service(s) by name."""
+        self._pebble.start_services(service_names)
+
+    def stop(self, *service_names: typing.List[str]):
+        """Stop given service(s) by name."""
+        self._pebble.stop_services(service_names)
+
+    def add_layer(self, setup: typing.Union[str, typing.Dict, 'pebble.Setup']):
+        """Dynamically add a service (Pebble) setup layer.
+
+        Args:
+            setup: A YAML string, setup dict, or pebble.Setup object
+                containing the Pebble setup layer to add.
+        """
+        if isinstance(setup, dict):
+            setup = pebble.Setup(setup)
+        self._pebble.add_layer(setup)
+
+    def get_setup(self) -> 'pebble.Setup':
+        """Fetch the flattened setup as a pebble.Setup object."""
```

Comment:
There are at least two ways out of this, I think: either we make this a layer API, or we turn setup into something that owns a list of layers which happens to have a single layer in this case. What do you think?

Reply:
Per above, I've gone the `pebble.Layer` route as I think that's what we need here.